src/helpers.rs

```diff
@@ -775,6 +775,24 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         this.write_scalar(scalar, &errno_place)
     }
 
+    fn set_libc_err_and_return_neg1(
+        &mut self,
+        err: &str,
+        dest: &MPlaceTy<'tcx>,
+    ) -> InterpResult<'tcx> {
+        let err = self.eval_libc(err);
+        self.set_last_err_and_return_neg1(err, dest)
+    }
+
+    fn set_last_err_and_return_neg1(
+        &mut self,
+        err: Scalar,
+        dest: &MPlaceTy<'tcx>,
+    ) -> InterpResult<'tcx> {
+        self.set_last_error(err)?;
+        self.eval_context_mut().write_int(-1, dest)
+    }
```

Comment:
Both of these new functions are superseded by the helper that @tiif added and that I extended. You should be able to use `set_last_error_and_return` now (after a rebase).